lute/static/js/dict-tabs.js

```diff
@@ -1,22 +1,61 @@
 "use strict";
 
+let dictTabs = [];
+
+class DictTab {
+  constructor(dict, frameName) {
+    const dictInfo = dict ? getDictInfo(dict) : null;
+    this.dictID = dictInfo ? dictInfo.id : null;
+
+    this.frame = this.createIFrame(frameName);
+    this.btn = document.createElement("button");
+    this.btn.classList.add("dict-btn");
+
+    if (this.dictID != null) {
+      this.label = dictInfo.label;
+      this.isExternal = dictInfo.isExternal;
+      this.btn.dataset.dictId = this.dictID;
+      this.btn.onclick = this.clickCallback.bind(this);
+      this.btn.dataset.dictExternal = this.isExternal ? "true" : "false";
+      
+      if (this.label != "") {
+        this.btn.textContent = this.label;
+        this.btn.setAttribute("title", this.label);
+      }
+
+      if (dictInfo.faviconURL) {
+        this.btn.prepend(createImg(dictInfo.faviconURL, "dict-btn-fav-img"));
+      }
+
+      if (this.isExternal) {
+        this.btn.appendChild(createImg("", "dict-btn-external-img"));
+      }
+    }
+  }
 
-function createTabBtn(label, dictID, external, faviconURL=null) {
-  const btn = document.createElement("button");
-  btn.classList.add("dict-btn");
-  if (label != "") {
-    btn.textContent = label;
-    btn.setAttribute("title", label);
+  clickCallback() {
+    if (this.isExternal) {
+      load_dict_popup(this.dictID);
+      return;
+    }
+    if (this.frame.dataset.contentLoaded == "false") {
```

Comment:
Following the dataset comment, this would become `if (!this.is_loaded())` or similar.